service/history/decisionChecker.go

```diff
@@ -170,8 +169,7 @@ func (c *workflowSizeChecker) failWorkflowSizeExceedsLimit() (bool, error) {
 			tag.WorkflowEventCount(historyCount))
 
 		attributes := &decisionpb.FailWorkflowExecutionDecisionAttributes{
-			Reason:  common.FailureReasonSizeExceedsLimit,
-			Details: payloads.EncodeString("Workflow history size / count exceeds limit."),
+			Failure: failure.NewServerFailure("Workflow history size / count exceeds limit.", false),
```

Comment:
Looks like you are breaking the pattern here from other server failures.  Is there a specific reason to treat these slightly differently.

Reply:
No, just overlooked. Moved message to constant.